### PR TITLE
Wizard Academy Away Update

### PR DESCRIPTION
**Pull request by AnturK**
The bulk of this is making user explicit in spells instead of inferred usr in every step now it's passed down the chain. The con is having to remember to add the user parameter in future spells even if it's unused. Want some feedback on this because i'm not entirely sure it's cleaner or not.

Simple mob wizard - Blinks,MMs and Fireballs. Nothing special besides that.

Academy Away changes:
- Gateway starts linked, removed scattered spawns
- Added simple wizards around the mission
- Special Spawner added in the headmaster room, it will spawn wizards and poll ghosts for control of them if players show up on the away mission.  The wizard has jaunt,mm,fireball and exile implant.

### CODE REVIEW

**Comment by tkdrg** [comment]
I think it's better this way, given that relying on usr is not safe (verbs aside), exploit/runtime prevention is always good.

**Comment by AnturK** [comment]
I probably should also remove the defaults to usr part, now it's supposed to be passed down.

**Comment by Robustin** [comment]
Haha, so ghosts can now murderbone away missioners?

On one hand making the academy more active is nice, on the other hand there's no reason to go on this mission (to my knowledge) if only death awaits and it might have the "syndie ship" effect of disappearing half the crew once the gateway opens.

**Comment by AnturK** [comment]
Yeah i need to come up with some rewards to add.

**Comment by KorPhaeron** [comment]
The reward is valids.

We need to redo exile implant code or people will just surgery them out.

**Comment by Zilenan91** [comment]
How about adding a charge spell in the headmasters office? Or a xeno egg in containment, or hell maybe a LMG if we want to be really insane and have the wizard at round start take all of the guns and murderbone with them.

**Comment by Bawhoppen** [comment]
simple animals are cancer

**Comment by KorPhaeron** [comment]
Zilenan 2/3 of those things are not wizard related

**Comment by Zilenan91** [comment]
Those were suggestions for what to put into the away mission, AnturK asked for them.

**Comment by Niknakflak** [comment]
Your ideas are bad zilenan

**Comment by Zilenan91** [comment]
Your face is bad

**Comment by Bawhoppen** [comment]
woah fucking edge-tier dunked

**Comment by Zilenan91** [comment]
Oh how about a wall safe in the headmasters office that has wands and maybe a spellbook of fireball or something. Would prevent the wizard from opening it if there's a code at round start, because I'm pretty sure this away mission is absent of tools.

**Comment by AnturK** [comment]
First pass of adding some silly loot.

**Comment by Zilenan91** [comment]
A veal render?

**Comment by duncathan** [comment]
Please

**Comment by KorPhaeron** [comment]
@duncathan 

its the one that summons cows silly

**Comment by Bawhoppen** [comment]
put a honk-render in there instead

**Comment by duncathan** [comment]
@KorPhaeron
The veal render is my #1 favorite item in the game

**Comment by Zilenan91** [comment]
The dice of fate sounds really interesting

File: code/modules/awaymissions/mission_code/Academy.dm [hunk 1]
```diff
@@ -40,4 +40,271 @@
...
+	head = /obj/item/clothing/head/bowler
+	glasses = /obj/item/clothing/glasses/monocle
+	gloves = /obj/item/clothing/gloves/color/white
+
+/obj/effect/proc_holder/spell/targeted/summonmob
+	name = "Summon Servant"
+	desc = "This spell can be used to call your servant, whenever you need it."
+	charge_max = 100
+	clothes_req = 0
+	invocation = "JE VES"
```
Comment (Dorsisdwarf) [inline code/modules/awaymissions/mission_code/Academy.dm]: I think I love you

**Comment by Zilenan91** [comment]
Oh hey, can you add a console or something to the "bridge" of the academy so that the wizard can send messages to the crew? He can challenge them to come fight him and steal his treasure so he isn't stuck there bored the whole shift.

**Comment by KorPhaeron** [comment]
@zilenan91

They dont spawn unless people go there

**Comment by AnturK** [comment]
How the hell this got merge conflicts.

**Comment by KorPhaeron** [comment]
Incomings spell balance stuff got merged

**Comment by KorPhaeron** [comment]
>smash window at start
>jetpack around to top
>smash windows at headmasters area
>let atmos vent
>every wizard who spawns will be suffocating for a full 30 seconds before a ghost takes control

Should probably do something about that

**Comment by KorPhaeron** [comment]
>You throw Dice of Fate. It lands on 2. 
>Talos Valcoran seizes up and falls limp, his eyes dead and lifeless...

Also fuck you

**Comment by Anonus** [comment]
You should stop spawning teachers after the first because holy shit round end report is full of wizards

**Comment by Zilenan91** [comment]
Nah more player wizards is hilarious. The windows should be unbreakable though, you shouldn't be able to vent this map.

**Pull request merged**